src/main/kotlin/net/lumalyte/lg/infrastructure/services/LumaGuildsChannel.kt

```
@@ -0,0 +1,194 @@
+ * per-player chat visibility toggles stored in [ChatSettingsRepository].
+ */
+class LumaGuildsChannel(provider: ChannelProvider) : RoseChatChannel(provider), KoinComponent {
+
+    private val memberService: MemberService by inject()
+    private val guildService: GuildService by inject()
+    private val relationService: RelationService by inject()
+    private val chatSettingsRepository: ChatSettingsRepository by inject()
+
+    /** Resolved from the `channel-type` key in `channels.yml` (default: GUILD). */
+    lateinit var channelType: LumaGuildsChannelType
+        private set
```

detekt (1.23.8)

[warning] Usages of lateinit should be avoided.

(detekt.potential-bugs.LateinitUsage)